presto-postgresql/src/main/java/com/facebook/presto/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -81,13 +89,35 @@ protected ResultSet getTables(Connection connection, String schemaName, String t
                 new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"});
     }
 
+    public static ReadMapping jsonReadMapping()
+    {
+        return sliceReadMapping(JSON, (resultSet, columnIndex) -> utf8Slice(resultSet.getString(columnIndex)));
```

Comment:
Looking at impl of `=` for JSON type (`com.facebook.presto.operator.scalar.JsonOperators#equals`), the value should be normalized. At least i would naively expect that `{'a': 1, 'b': 2} = {'b': 2, 'a': 1}`.

You probably need to do something like `cast(some_varchar as json)` is doing -- see `com.facebook.presto.operator.scalar.JsonOperators#castFromVarchar`. Probably the impl of the cast should be moved to utility method (somewhere) and reused.

@electrum json is not part of SPI. How do you think this should be addressed?

Reply:
We should be able to do this with `TypeManager` from `ConnectorContext` by resolving the `CAST` operator. This will require a small change to `resolveOperator()` (or adding a new method ... need to get input from @martint and others on this):

```java
@Override
public MethodHandle resolveOperator(OperatorType operatorType, List<? extends Type> argumentTypes)
{
    requireNonNull(functionRegistry, "functionRegistry is null");
    Signature signature;
    if (operatorType == OperatorType.CAST) {
        checkArgument(argumentTypes.size() == 2, "CAST requires exactly two arguments");
        signature = functionRegistry.getCoercion(argumentTypes.get(0), argumentTypes.get(1));
    }
    else {
        signature = functionRegistry.resolveOperator(operatorType, argumentTypes);
    }
    return functionRegistry.getScalarFunctionImplementation(signature).getMethodHandle();
}